This notebook runs a test to check whether costs are incurred, and by who, for the download of a file from the Seven Bridges CGC instance.

The file is a 33GB sam file.

In [1]:
import json

from fasp.loc import sbcgcDRSClient
drsClient = sbcgcDRSClient('/Users/forei/.keys/sbcgc_key.json', 's3')
drs_id ='62ac9f0ef08fea477043aa0f'
drs_object = drsClient.get_object(drs_id)
print(json.dumps(drs_object,indent=3))
file_size_bytes = drs_object['size']
file_size_gb = file_size_bytes/1024**3
print("File size {} GB".format(file_size_gb))

{
   "id": "62ac9f0ef08fea477043aa0f",
   "name": "HCC1143_1.sam",
   "size": 35671819315,
   "checksums": [
      {
         "type": "etag",
         "checksum": "bf036831cd957fa76b2cf8e16258a834-34"
      }
   ],
   "self_uri": "drs://cgc-ga4gh-api.sbgenomics.com/62ac9f0ef08fea477043aa0f",
   "created_time": "2022-06-17T15:34:38Z",
   "updated_time": "2022-06-17T15:34:38Z",
   "mime_type": "application/json",
   "access_methods": [
      {
         "type": "s3",
         "region": "us-east-1",
         "access_id": "aws-us-east-1"
      }
   ]
}
File size 33.221970605663955 GB


#### Download cost and who is charged to
Checking the billing on the account before and after download the 33.2 Gb file cost $2.99 to download.

In this case it was ambiguous whether it was the owner or the downloader to whom accesa was charged. The same account was used for both. Need a better designed test!